coordinator/persistence-common/src/main/java/io/stargate/db/datastore/common/util/UserDefinedFunctionHelper.java

```diff
@@ -39,18 +40,32 @@ public static void fixCompilerClassLoader() {
       Class<?> targetClass =
           Class.forName("org.apache.cassandra.cql3.functions.UDFunction$UDFClassLoader");
       Field targetField = targetClass.getDeclaredField("insecureClassLoader");
-      targetField.setAccessible(true);
 
-      Field modifiers = Field.class.getDeclaredField("modifiers");
-      modifiers.setAccessible(true);
-      modifiers.setInt(targetField, targetField.getModifiers() & ~Modifier.FINAL);
-
-      targetField.set(null, targetClass.getClassLoader());
+      // As per https://stackoverflow.com/questions/74723932/java-17-reflection-issue
+      // we must use more complicated handling on JDK 12+
```

Comment:
... alas, does not seem to work on JDK 17 so instead remove calls altogether.

Reply:
But as per #2788, commenting out breaks DSE-6.8, DSE-next and C-4.0 backends. Issue, whatever it is, is indeed fixed in C-4.1.